Plugins/GitUIPluginInterfaces/IGitPlugin.cs

```diff
@@ -28,4 +28,34 @@ public interface IGitPlugin
         /// </summary>
         bool Execute(GitUIEventArgs args);
     }
-}
\ No newline at end of file
+
+    public interface IGitPluginV2
+    {
+        string Name { get; }
+
+        string Description { get; }
+
+        [CanBeNull]
+        Image Icon { get; }
+
+        /// <summary>
+        /// Runs the plugin and returns whether the RevisionGrid should be refreshed.
+        /// </summary>
+        bool Execute(GitUIEventArgs args);
+    }
+
+    public interface IGitPluginRegistrable
+    {
+        void Register(IGitUICommands gitUiCommands, IGitPluginSettingsContainer settingsContainer);
+    }
+
+    public interface IGitPluginUnregistrable
+    {
+        void Unregister();
+    }
```

Comment:
This pair of functions should be in one interface, shouldn't it?

> The purpose is not entirely clear for Register and Unregister.
It is similar to loading / unloading events.
I would like to rename these methods.

The interfaces should have a documentation as a basic principle.
These functions are called by the Plugin**Registry**. Though "Load / Unload" might be more clear anyway.

Reply:
> This pair of functions should be in one interface, shouldn't it?

Not all plugins use both of them.

Reply:
> The interfaces should have a documentation as a basic principle.

Added documentation for interfaces.